code/modules/vehicles/animal.dm

```diff
@@ -76,7 +84,11 @@
 			to_chat(load, "You were unbuckled from \the [src] by [user]")
 
 /obj/vehicle/animal/bullet_act(var/obj/item/projectile/Proj)
-	if(buckled && prob(protection_percent))
+	var/datum/component/armor/armor_component = GetComponent(/datum/component/armor)
+	var/key = get_armor_key(Proj.damage_type, Proj.damage_flags)
+	if(armor_component.get_value(key))
+		block_chance = armor_component.get_value(key)
+	if(buckled && prob(block_chance))
```

Comment:
instead of all this, use get_blocked on the armour component to get the blocked ratio here (result will be from 0 to 1)

Reply:
Done